system/jhtree/ctfile.cpp

```diff
@@ -24,12 +24,17 @@
 #endif
 
 #include "jmisc.hpp"
+#include "jset.hpp"
 #include "hlzw.h"
 
 #include "ctfile.hpp"
 #include "jstats.h"
 
-void SwapBigEndian(KeyHdr &hdr)
+//#define TIME_NODE_SEARCH            // Useful for performance testing, but generally should be off
+
+//---------------------------------------------------------------------------------------------------------------------
+
+void SwapBigEndian(KeyHdr &hdr, bool read)
```

Comment:
i don't like having swapBigEndian also do compatibility patching. Better to leave if without the parameter and do the compatibility patch elsewhere (or don't do it at all - not really sure why it is needed, given we could just add new node types)